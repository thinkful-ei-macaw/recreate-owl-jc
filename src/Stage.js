import React from 'react'
import Tile from './Tile';

const Stage = ({participants}) => {
  return (
    <div>
    {
      participants.map(participant => participant.onStage ? 
        <Tile  
        key={participant.id}
        avatar={participant.avatar}
        name={participant.name}
        />:null)
    }
    </div>
  )
}

export default Stage;