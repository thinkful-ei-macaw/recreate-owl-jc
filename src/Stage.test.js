import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stage from './Stage';

describe('Stage component tests', () => {
  // smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Stage participants={[
      {
          id: 1,
          name: 'Koren Templeton',
          avatar:
              'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: true
      },
      {
          id: 2,
          name: 'Caty Flucker',
          avatar:
              'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: false
      }]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  // snapshot test
  it('renders Stage with participant info', () => {
    const tree = renderer
    .create(<Stage participants={[
      {
          id: 1,
          name: 'Koren Templeton',
          avatar:
              'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: true
      },
      {
          id: 2,
          name: 'Caty Flucker',
          avatar:
              'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: false
      }]}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })
})