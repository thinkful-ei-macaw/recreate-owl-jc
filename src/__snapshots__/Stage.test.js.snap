// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage component tests renders Stage with participant info 1`] = `
<div>
  <div>
    <img
      alt="Avatar"
      className="tile-image"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <p>
      Koren Templeton
    </p>
  </div>
</div>
`;
